src/plugins/packageManager.js

```diff
@@ -0,0 +1,113 @@
+const packagesConfig = path.join(packagesParent, "package.json");
+const packagesNpmrc = path.join(packagesParent, ".npmrc");
+
+const packageDirJson = {
+	private: true,
+	description: "Packages for The Lounge. All packages in node_modules directory will be automatically loaded.",
+};
+const npmrc = {
+	save: false,
+	"bin-links": false,
+	progress: false,
+	"package-lock": false,
+	production: true,
+};
+
+module.exports = {
+	install,
+	uninstall,
+};
+
+function checkConfig() {
+	return fs.access(Helper.getConfigPath(), fs.constants.R_OK | fs.constants.W_OK)
+		.catch(() => Promise.reject(`${Helper.getConfigPath()} does not exist or is not readable.`));
+}
+
+function getPackageJson(packageName) {
+	log.info("Retrieving information about the package...");
+	return packageJson(packageName, {
+		fullMetadata: true,
+	});
+}
+
+function getMetadata(json) {
+	if (!("thelounge" in json)) {
+		return Promise.reject(`${colors.red(json.name)} does not have The Lounge metadata.`);
+	}
+	const metadata = json.thelounge;
+	metadata.version = json.version;
+}
+
+function execNpm(command, packageName, metadata) {
+	return new Promise((res, rej) => {
+		const npm = child.spawn(
+			process.platform === "win32" ? "npm.cmd" : "npm",
+			[
+				command,
+				"--prefix",
+				packagesParent,
+				packageName,
+			],
+			{
+				// This is the same as `"inherit"` except `process.stdout` is ignored
+				stdio: [process.stdin, "ignore", process.stderr],
+			}
+		);
+
+		npm.on("error", rej);
+
+		npm.on("close", (code) => {
+			if (code !== 0) {
+				return rej(`Failed to ${command} ${colors.green(`${packageName} " v" + ${metadata.version}`)}. Exit code: ${code}`);
+			}
+			res();
+		});
+	});
+}
+
+function buildNpmrcString() {
+	let string = "";
+	Object.keys(npmrc).forEach((key) => string += `${key} = ${npmrc[key]}\n`);
+	return string;
+}
+
+function runNpmCommand(command, packageName, metadata) {
+	log.info(`${command}ing ${colors.green(packageName)}...`);
+
+	return fs.ensureDir(packagesPath) // Create node_modules folder, otherwise npm will start walking upwards to find one
+		.then(() => fs.writeJson(packagesConfig, packageDirJson, {spaces: "\t"})) // Create package.json with private set to true to avoid npm warnings
+		.then(() => fs.writeFile(packagesNpmrc, buildNpmrcString()))
+		.then(execNpm.bind(this, command, packageName, metadata));
+}
+
+function install(packageName) {
+	checkConfig()
+		.then(getPackageJson.bind(this, packageName))
+		.then(getMetadata)
+		.then(runNpmCommand.bind(this, "install", packageName))
+		.then(() => log.info(`${colors.green(packageName)} has been successfully installed.`))
+		.catch((e) => {
+			log.error(`${e}`);
+			process.exit(1);
+		});
+}
+
+function uninstall(packageName) {
+	checkConfig()
+		.then(runNpmCommand.bind(this, "uninstall", packageName))
+		.then(() => log.info(`${colors.green(packageName)} has been successfully uninstalled.`))
+		.catch((e) => {
+			log.error(`${e}`);
+			process.exit(1);
+		});
```

Comment:
This currently loses some functionality: if the package was not installed, it now says "... has been successfully uninstalled". I know this is normal npm behavior, but I added that on purpose, it's good to know that a package was actually not uninstalled, which is why I had the warning.

Also, it now creates the `package.json` file and package folder when running uninstall without having installed anything. I don't have anything against that, but that's something @xPaw [made me change](https://github.com/thelounge/lounge/pull/1938#discussion_r159850803) 😁 Any reason why we're going back to that? Just for code convenience?

Reply:
Good catch, I shall add a check to see if the plugin is installed before trying to uninstall it.

Reply:
Hmmm, so I was having troubles actually making this work the same way you did it. It turns out, yours doesn't work. npm still outputs something even when it didn't uninstall. So I'm now checking against the output to see whether we uninstalled or not. When not uninstalling it outputs "up to date" and when uninstalling successfully it outputs "removed 1 package". Did you manage to get this working locally, is there perhaps some setting you forgot to push?

> Also, it now creates the package.json file and package folder when running uninstall without having installed anything. I don't have anything against that, but that's something @xPaw made me change 😁 Any reason why we're going back to that? Just for code convenience?

Nope, I just hadn't realised yours didn't do that. I shall remove it :-)

Reply:
Oh boy...

<img width="518" alt="screen shot 2018-01-13 at 01 15 48" src="https://user-images.githubusercontent.com/113730/34903426-55375c72-f7ff-11e7-8e01-f2eeb81a3de4.png">

Reply:
For the same reason, you can't rely on keywords `removed` or `up-to-date`. Otherwise you'll keep fixing on v8 and breaking v6, then I'll stop by and fix v6, breaking v8. Repeat 😂

Reply:
So, I have spent a rather extensive amount of time trying to please nvm v3 to v5 based on output of `npm uninstall` and no luck. In the end, I fixed that more cleanly in https://github.com/thelounge/lounge/pull/1974.
Yeah, it's an extra step, but it's worth it for robustness. We are less subject to npm variations (we still rely on the package name to be part in the output, but that's about it as opposed to my previous version and yours :man_shrugging:).

Also... sorry for the rebase :/